spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -55,6 +56,7 @@ public Channel createChannel(boolean transactional) {
 		try {
 			Channel channel = this.delegate.createChannel();
 			if (transactional) {
+				Assert.state(channel != null, "Can't start the transaction  - channel is null.");
```

Comment:
I think the assertion should be before the if, with a message "Client returned a null channel - perhaps maxChannels exceeded?".

Reply:
At first I had the same idea as you.But I worry about affecting other parts.

Reply:
We would just get NPEs elsewhere if we let a `null` channel escape from here.